controllers/replication-controller/dellcsireplicationgroup_controller.go

```diff
@@ -478,3 +511,319 @@ func (r *ReplicationGroupReconciler) SetupWithManager(mgr ctrl.Manager, limiter
+		return err
+	}
+
+	// Update the target PV's claimref to point to our pvc.
+	pvcUID := pvc.ObjectMeta.UID
+	pvcResourceVersion := pvc.ObjectMeta.ResourceVersion
+	err = updatePVClaimRef(ctx, client, targetPV, pvc.Namespace, pvcResourceVersion, pvc.Name, pvcUID, log)
+	if err != nil {
+		return err
+	}
+
+	// Verify pvc is created and bound to new PVs
+	// remotePV is the current localPVName arg, localPV is the current remotePVName arg
+	err = verifyPVC(ctx, client, remotePV, localPV, pvcName, namespace, log)
+	if err != nil {
+		return err
+	}
+
+	// Remove the PVC reclaim of local PV
+	err = removePVClaimRef(ctx, client, localPV, namespace, pvcName, log)
+	if err != nil {
+		return err
+	}
+
+	// Restore the PVs original volume reclaim policy
+	err = setPVReclaimPolicy(ctx, client, pvc.Spec.VolumeName, remotePVPolicy, log)
+	if err != nil {
+		return err
+	}
+	err = setPVReclaimPolicy(ctx, client, pvc.Annotations[controller.RemotePV], localPVPolicy, log)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func verifyPVC(ctx context.Context, client connection.RemoteClusterClient, localPVName string, remotePVName string, pvcName string, namespace string, log logr.Logger) error {
+	for iteration := 0; iteration < 30; iteration++ {
+		pvc, err := client.GetPersistentVolumeClaim(ctx, namespace, pvcName)
+
+		if (err == nil) && (localPVName == pvc.Spec.VolumeName) && (remotePVName == pvc.Annotations[controller.RemotePV]) {
+			return nil
+		}
+
+		time.Sleep(2 * time.Second)
+	}
+	log.V(common.InfoLevel).Info(fmt.Sprintf("Timed out waiting on PVC %s/%s to be created and bound", namespace, pvcName))
+	return fmt.Errorf("timed out waiting on PVC %s/%s to be created and bound", namespace, pvcName)
+}
+
+func setPVReclaimPolicy(ctx context.Context, client connection.RemoteClusterClient, pvName string, prevPolicy v1.PersistentVolumeReclaimPolicy, log logr.Logger) error {
+	for iteration := 0; iteration < 30; iteration++ {
+		pv, err := client.GetPersistentVolume(ctx, pvName)
+		if err != nil {
+			log.V(common.InfoLevel).Info(fmt.Sprintf("Error retrieving PV %s: %s", pvName, err.Error()))
+			return err
+		}
+		pv.Spec.PersistentVolumeReclaimPolicy = prevPolicy
+		err = client.UpdatePersistentVolume(ctx, pv)
+		if err != nil {
+			log.V(common.InfoLevel).Info(fmt.Sprintf("Error updating PV %s: %s", pvName, err.Error()))
+			return err
+		}
+		pv, err = client.GetPersistentVolume(ctx, pvName)
+		if err != nil {
+			log.V(common.InfoLevel).Info(fmt.Sprintf("Error retrieving PV %s: %s", pvName, err.Error()))
+			return err
+		}
+		if pv.Spec.PersistentVolumeReclaimPolicy == prevPolicy {
+			return nil
+		}
+
+		time.Sleep(2 * time.Second)
+	}
+	err := fmt.Errorf("timed out waiting on PV VolumeReclaimPolicy to be set to previous policy")
+	return err
+}
+
+func makePVReclaimPolicyRetain(ctx context.Context, client connection.RemoteClusterClient, pvName string, log logr.Logger) error {
```

Comment:
this function and `setPVReclaimPolicy` are essentially the same function. 
Have you considered calling `setPVReclaimPolicy` with "Retain" instead of creating this new func?

Reply:
@santhoshatdell covered this above.